test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Data;
+using System.Data.SqlClient;
+using Microsoft.ML.Data;
+using Microsoft.ML.TestFramework;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class DatabaseLoaderTests : BaseTestClass
+    {
+        public DatabaseLoaderTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void TryItOut()
+        {
+            var mlContext = new MLContext(seed: 1);
+
+            var connectionString = @"Server=(localdb)\mssqllocaldb;Database=EFGetStarted.AspNetCore.NewDb;Trusted_Connection=True;ConnectRetryCount=0";
```

Comment:
We will either need to disable this test, or change it so it doesn't rely on machine state/configuration.

Reply:
I created a `MockDbConnection`, `MockDbCommand`, and `MockDbDataReader` class which wrap a `TextLoader`. I would guess this needs some cleanup to support more testing scenarios and to match w/e conventions ML.NET is currently using.